File: backend/components/metiers/administration_publique/guides/helpers/plugins/helpers_plugins.test.js
// helpers_plugins.test.js – Tests unitaires et edge cases JS
const { checkPlugin, auditPlugin } = require('./helpers_plugins');

test('checkPlugin returns true for enabled', () => {
  expect(checkPlugin({ enabled: true, version: '1.0.0' })).toBe(true);
  expect(checkPlugin({ enabled: false, version: null })).toBe(false);
});

test('auditPlugin returns audit object', () => {
  const result = auditPlugin({ enabled: true, version: '1.0.0' });
  expect(typeof result).toBe('object');
  expect(result).toHaveProperty('score');
  expect(result).toHaveProperty('compliant');
});

File: backend/components/metiers/administration_publique/guides/helpers/plugins/helpers_plugins.test.py
# helpers_plugins.test.py – Tests unitaires et edge cases Python
from .helpers_plugins import check_plugin, audit_plugin

def test_check_plugin():
    assert check_plugin({'enabled': True, 'version': '1.0.0'}) is True
    assert check_plugin({'enabled': False, 'version': None}) is False

def test_audit_plugin():
    result = audit_plugin({'enabled': True, 'version': '1.0.0'})
    assert isinstance(result, dict)
    assert 'score' in result
    assert 'compliant' in result
